cmd/bgpfinder-server/main.go

```diff
@@ -0,0 +1,183 @@
+
+	projects, err := bgpfinder.Projects()
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error fetching projects: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	if projectName == "" {
+		// Return all projects
+		jsonResponse(w, projects)
+	} else {
+		// Return specific project if exists
+		for _, project := range projects {
+			if project.Name == projectName {
+				jsonResponse(w, project)
+				return
+			}
+		}
+		http.Error(w, "Project not found", http.StatusNotFound)
+	}
+}
+
+// collectorHandler handles /meta/collectors and /meta/collectors/{collector} endpoints
+func collectorHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	collectorName := vars["collector"]
+
+	collectors, err := bgpfinder.Collectors("")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error fetching collectors: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	if collectorName == "" {
+		// Return all collectors
+		jsonResponse(w, collectors)
+	} else {
+		// Return specific collector if exists
+		for _, collector := range collectors {
+			if collector.Name == collectorName {
+				jsonResponse(w, collector)
+				return
+			}
+		}
+		http.Error(w, "Collector not found", http.StatusNotFound)
+	}
+}
+
+// parseDataRequest parses the HTTP request and builds a bgpfinder.Query object
+func parseDataRequest(r *http.Request) (bgpfinder.Query, error) {
+	query := bgpfinder.Query{}
+
+	intervalsParams := r.URL.Query()["intervals[]"]
+	collectorsParams := r.URL.Query()["collectors[]"]
+	typesParams := r.URL.Query()["types[]"]
+
+	// Parse interval
+	if len(intervalsParams) == 0 {
+		return query, fmt.Errorf("at least one interval is required")
+	}
+
+	times := strings.Split(intervalsParams[0], ",")
+	if len(times) != 2 {
+		return query, fmt.Errorf("invalid interval format. Expected format: start,end")
+	}
+
+	startInt, err := strconv.ParseInt(times[0], 10, 64)
+	if err != nil {
+		return query, fmt.Errorf("invalid start time: %v", err)
+	}
+
+	endInt, err := strconv.ParseInt(times[1], 10, 64)
+	if err != nil {
+		return query, fmt.Errorf("invalid end time: %v", err)
+	}
+
+	query.From = time.Unix(startInt, 0)
+	query.Until = time.Unix(endInt, 0)
+
+	// Parse collectors
+	var collectors []bgpfinder.Collector
+	if len(collectorsParams) == 0 {
+		// Use all collectors
+		collectors, err = bgpfinder.Collectors("")
+		if err != nil {
+			return query, fmt.Errorf("error fetching collectors: %v", err)
+		}
+	} else {
+		// Use specified collectors
+		allCollectors, err := bgpfinder.Collectors("")
+		if err != nil {
+			return query, fmt.Errorf("error fetching collectors: %v", err)
+		}
+
+		collectorMap := make(map[string]bgpfinder.Collector)
+		for _, c := range allCollectors {
+			collectorMap[c.Name] = c
+		}
+
+		for _, name := range collectorsParams {
+			if collector, exists := collectorMap[name]; exists {
+				collectors = append(collectors, collector)
+			} else {
+				return query, fmt.Errorf("collector not found: %s", name)
```

Comment:
I'm on the fence about whether we should error out here or just silently drop the collector. Let's keep it like this for now, but something to think about.